### PR TITLE
Refactor to improve types and docs for `*-list` rules

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Follow-up of #5934

> Is there anything in the PR that needs further explanation?

When I was looking into the problem of #5933 and #5934, I felt to check the `*-list` rules again.

So, this pull request aims to improve the docs and type of that rules for consistency and clarity.
It doesn't change the existing behavior.

### CODE REVIEW

File: lib/rules/at-rule-allowed-list/__tests__/index.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ testRule({
 	ruleName,
 	skipBasicChecks: true,
 
-	config: ['keyframes'],
+	config: 'keyframes',
```
Comment (ybiquitous) [inline lib/rules/at-rule-allowed-list/__tests__/index.js:83]: [note] This change aims to test a single primary option value.

File: lib/rules/at-rule-allowed-list/index.js [hunk 2]
```diff
@@ -17,23 +17,19 @@ const meta = {
 	url: 'https://stylelint.io/user-guide/rules/list/at-rule-allowed-list',
 };
 
-/** @type {import('stylelint').Rule} */
+/** @type {import('stylelint').Rule<string | string[]>} */
 const rule = (primary) => {
-	// To allow for just a string as a parameter (not only arrays of strings)
-	const primaryValues = [primary].flat();
```
Comment (ybiquitous) [inline lib/rules/at-rule-allowed-list/index.js:23]: [note] `validateOptions` allows both a single value and an array, so it's better to call `.flat()` after the validation.

File: lib/rules/selector-combinator-disallowed-list/__tests__/index.js [hunk 3]
```diff
@@ -64,3 +64,21 @@ testRule({
...
+		},
+	],
+
+	reject: [
+		{
+			code: 'a > b {}',
+			message: messages.rejected('>'),
+		},
+	],
+});
```
Comment (ybiquitous) [inline lib/rules/selector-combinator-disallowed-list/__tests__/index.js:84]: [note] This new test case aims to just check a single primary value.

File: lib/utils/containsString.js [hunk 4]
```diff
@@ -33,10 +33,8 @@ module.exports = function containsString(input, comparison) {
 };
 
 /**
- *
  * @param {string} value
- * @param {string} comparison
- *
+ * @param {unknown} comparison
```
Comment (ybiquitous) [inline lib/utils/containsString.js:37]: [note] `testAgainst()` uses `isString()`, so it accepts the `unknown` argument.

**Review by jeddy3 — APPROVED** [review]
Fantastic cleanup and refactor, thank you!

LGTM.

**Pull request merged**